### PR TITLE
Source Jira - Pagination fix on jira users stream

### PR DESCRIPTION
**Pull request by antonioneto-hotmart**
## What
This change fixes an existing pagination bug in jira source user stream, currently it only returns 50 results by default

## How
A field has been added to the interface, so that the user can define the amount of results to be loaded

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
This is a simple change

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/source_jira/spec.json [hunk 1]
```diff
@@ -25,6 +25,11 @@
         "title": "Email",
         "description": "The user email for your Jira account."
       },
+      "max_results": {
+        "type": "string",
+        "title": "Max Results",
+        "description": "Pagination max results (only for users stream)"
```
Comment (koconder) [inline airbyte-integrations/connectors/source-jira/source_jira/spec.json:31]: Few things:
1. State in the description default is 50
2. Set type to `number`

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 2]
```diff
@@ -1083,8 +1083,14 @@ class Users(JiraStream):
...
 
+    def __init__(self, domain: str, projects: List[str], max_results: str, **kwargs):
+        super(JiraStream, self).__init__(**kwargs)
+        self._domain = domain
+        self._projects = projects
+        self._max_results = max_results
+
     def path(self, **kwargs) -> str:
-        return "user/search?query="
+        return "user/search?maxResults=" + self._max_results + "&query="
```
Comment (koconder) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:1093]: if `maxResults` number is null/empty/0 then do not set in the return

**Review by vincentkoc — CHANGES_REQUESTED** [review]
Hi @antonioneto-hotmart I will be reviewing your PR. Thanks again for your contribution. Few small changes, see specific comments. Mostly about handling when the value is not set.

Additionally:

1. You will also need to include your integration/unit tests in the PR to validate the rest of the code is working as expected
2. Bump version in the image here https://github.com/airbytehq/airbyte/blob/7147e1799b5c911bba9767916633c1b1a1a64456/airbyte-integrations/connectors/source-jira/Dockerfile#L15 to `0.2.22`
3. Changelog updated in docs/integrations/<source or destination>/<name>.md including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog) - Link: https://github.com/airbytehq/airbyte/blob/7147e1799b5c911bba9767916633c1b1a1a64456/docs/integrations/sources/jira.md?plain=1#L98 as version `0.2.22`

**Comment by sajarin** [comment]
Hey @antonioneto-hotmart, any chance you've gotten an opportunity to view the comments above? We've love to get this merged!

**Comment by koconder** [comment]
@antonioneto-hotmart let me know if you need any help

**Comment by koconder** [comment]
@antonioneto-hotmart the source.py changes are throwing the final error:
```
File \"/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 72, in discover\n    streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]\n  File \"/airbyte/integration_code/source_jira/source.py\", line 104, in streams\n    users_args = *****args, \"max_results\": config[\"max_results\"]***\nKeyError: 'max_results'\n", "failure_type": "system_error"***
```

**Comment by koconder** [comment]
@antonioneto-hotmart same issue again, the main problem is these files are missing the values for max_results:
* https://github.com/airbytehq/airbyte/blob/c97eac355edd8f703f5960e9d14f17b601bbe3d1/airbyte-integrations/connectors/source-jira/integration_tests/sample_config.json
* https://github.com/airbytehq/airbyte/blob/c97eac355edd8f703f5960e9d14f17b601bbe3d1/airbyte-integrations/connectors/source-jira/integration_tests/invalid_config.json

However like start date, this should be optional but your code is explicitly expecting a response which for the unit tests are coming from the two JSON files above.

**Comment by antonioneto-hotmart** [comment]
> However like start date, this should be optional but your code is explicitly expecting a response which for the unit tests are coming from the two JSON files above.

Would it be correct to add it to the two .json (becoming obligatory, as I understood), since it has a standard value (50)? This value that will be supplied in any case if it is not prefilled (being optional)

**Comment by antonioneto-hotmart** [comment]
@koconder I credit that now you are all right

**Comment by antonioneto-hotmart** [comment]
> @antonioneto-hotmart i didn’t check the code deeply, took another look and found the issue and corrected. I'll get the changes published now :)

@koconder thank you very much for the support!

**Review by vincentkoc — APPROVED** [review]

**Comment by antonioneto-hotmart** [comment]
Hey @koconder, do you know why I haven't followed merged in master yet?

**Comment by sajarin** [comment]
Thanks @antonioneto-hotmart for the PR and thanks @koconder for the review!

**Comment by antonioneto-hotmart** [comment]
@sajarin It seems to have given an error in the commit pipe, do you know what could have happened? Sorry, I'm new to open source:
Captura de Tela 2022-10-24 às 17 30 34

**Pull request merged**